src/future.rs

```diff
@@ -0,0 +1,75 @@
+use std::future::Future;
+use std::pin::Pin;
+use std::sync::{Arc, Mutex};
+use std::task::{Context, Poll, Waker};
+
+struct GpuFutureInner<T> {
```

Comment:
We must be careful with naming here: a lot of asynchronouity/futures come from the fact that there is a browser and potential IPC boundary to the GPU process (on the CPU side), not necessarily GPU itself.

Reply:
Suggestions re name? Right now it's really just Java's `CompletableFuture` plus a little bit of logic so the user doesn't have to remember to  invoke `wgpu_device_poll` manually in single-threaded scenarios.

Reply:
ok, I don't have other naming candidates :)